js/modals/question-modal.js

```diff
@@ -3,60 +3,53 @@ $(function() {
 
    function hintButtonClick($button) {
       if (!$button.hasClass('active')) {
-         console.log('hint click, not active');
-         console.log('data hint: ' + $button.data('hint'));
-
-         $('fieldset.hints .hint.active').data('hint-text', $('fieldset.hints textarea').val());
-         $('fieldset.hints textarea').val($button.data('hint-text'));
-         $('fieldset.hints textarea').attr('placeholder', 'Hint ' + $button.data('hint'));
-         $('.hint.active>span').text($('.hint.active').data('hint'));
-         $('.hint.active').removeClass('active');
+         $('.modal fieldset.hints .hint.active').data('hint-text', $('.modal fieldset.hints textarea').val());
```

Comment:
This selector looks very over-specified. Let's reduce it to something smaller. As the the first thing, you can remove `fieldset.hints`.